src/App.js

```diff
@@ -1,85 +1,163 @@
 import React from 'react';
+import TodoList from './components/TodoList';
+import TodosFilter from './components/TodosFilter';
+import NewTodo from './components/NewTodo';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main">
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">asdfghj</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">qwertyuio</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">zxcvbnm</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-4" />
-              <label htmlFor="todo-4">1234567890</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+const filterTypes = ['All', 'Active', 'Completed'];
+
+const filterTodos = (todos, filter) => {
+  if (filter === 'Active') {
+    return todos.filter(todo => todo.completed === false);
+  }
+
+  if (filter === 'Completed') {
+    return todos.filter(todo => todo.completed === true);
+  }
+
+  return todos;
+};
+
+class App extends React.Component {
+  state = {
+    todos: [],
+    visibleTodos: [],
+    filter: 'All',
+  }
+
+  addTodo = (todo) => {
+    const { filter, todos } = this.state;
+
+    this.setState(() => {
+      const allTodos = [...todos, todo];
+
+      return {
+        todos: allTodos,
+        visibleTodos: filterTodos(allTodos, filter),
```

Comment:
call this function in render method and store the value to some variable that you will pass to todolist so that you don't have two lists in the state

Reply:
done